### PR TITLE
#15391 Enhance Docker images to comply with the security best practice from AWS Marketplace

### PR DESCRIPTION
**Pull request by michael-jong-ds**
1. change image pull template to use imageCredentials registry if provided
2. add service account for pod identity/IRSA to backend pods

### CODE REVIEW

File: charts/datasaur/templates/config/_image-pull-secret.tpl [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{- define "imagePullSecret" }}
-{{- printf "{\"auths\": {\"%s\": {\"auth\": \"%s\"}}}" .Values.image.registry (printf "%s:%s" .Values.imageCredentials.username .Values.imageCredentials.password | b64enc) | b64enc }}
+{{- printf "{\"auths\": {\"%s\": {\"auth\": \"%s\"}}}" (default .Values.image.registry .Values.imageCredentials.registry) (printf "%s:%s" .Values.imageCredentials.username .Values.imageCredentials.password | b64enc) | b64enc }}
```
Comment (tonoman3g) [inline charts/datasaur/templates/config/_image-pull-secret.tpl:2]: before this PR, have we supported changing image registry? 🤔  why do you need to add the registry again here? @michael-jong-ds

File: charts/datasaur/values.yaml [hunk 2]
```diff
@@ -353,4 +354,7 @@ elasticsearch:
   kibana:
     enabled: false
   storage: 100Gi
-  nodeCount: 3
\ No newline at end of file
+  nodeCount: 3
+
+clusterSettings:
```
Comment (tonoman3g) [inline charts/datasaur/values.yaml:359]: could you elaborate more about the irsa / pod identity here as comment?

File: charts/datasaur/templates/config/_image-pull-secret.tpl (hunk 1 above)
Comment (daniel-jeremia) [reply charts/datasaur/templates/config/_image-pull-secret.tpl:2]: ini ada issue kemarin..
Jadi .Values.image.registry itu kan <aws-id>.us.aws.etc/datasaur/test
Nah tapi credentialsnya itu harus di assign ke <aws-id>.us.aws.etc saja tanpa /datasaur/test, jadi aku saranin mike buat dipisah @tonoman3g

File: charts/datasaur/values.yaml (hunk 2 above)
Comment (daniel-jeremia) [reply charts/datasaur/values.yaml:359]: Have u tested if the IRSA becomes instance profile like ec2? @michael-jong-ds

Comment (michael-jong-ds) [reply charts/datasaur/values.yaml:359]: how do I check that ya niel?
in env there is this env from irsa
Screenshot from 2024-03-06 10-44-17

File: charts/datasaur/templates/config/_image-pull-secret.tpl (hunk 1 above)
Comment (michael-jong-ds) [reply charts/datasaur/templates/config/_image-pull-secret.tpl:2]: this is to pull the image from the provided repository in values.yaml, or if the value is not there, use the default image registry

File: charts/datasaur/values.yaml (hunk 2 above)
Comment (daniel-jeremia) [reply charts/datasaur/values.yaml:359]: oh bisa pake aws sts get-called-identity aja mike

Comment (michael-jong-ds) [reply charts/datasaur/values.yaml:359]: ohh iya kalo sts get-caller-identity sudah pakai arn service account irsa nya
```
{
    "UserId": "AROA6ODU75TFBCDY4G4CN:botocore-session-1709715567",
    "Account": "992382807242",
    "Arn": "arn:aws:sts::992382807242:assumed-role/eksIRSARole/botocore-session-1709715567"
}
```

Comment (daniel-jeremia) [reply charts/datasaur/values.yaml:359]: oh bisa pake aws sts get-called-identity aja mike

**Review by daniel-jeremia — APPROVED** [review]

**Review by tonoman3g — APPROVED** [review]

**Pull request merged**